block_evasys_sync.php

```diff
@@ -51,202 +47,40 @@ public function get_content() {
-
-            // Only use standardtime js if no record exists.
-            if (!$record) {
-                $this->page->requires->js_call_amd('block_evasys_sync/standardtime', 'init');
-            }
-        }
-        $evasyssynchronizer = new \block_evasys_sync\evasys_synchronizer($this->page->course->id);
-        try {
-            $evasyscourses = $evasyssynchronizer->get_allocated_courses();
-        } catch (Exception $exception) {
-            \core\notification::warning(get_string('syncnotpossible', 'block_evasys_sync'));
-            $this->content->text .= html_writer::div(get_string('syncnotpossible', 'block_evasys_sync'));
-            return $this->content;
-        }
-
-        if ($ismodeautomated) {
-            $href = new moodle_url('/course/view.php',
-                                   array('id' => $this->page->course->id, "evasyssynccheck" => true));
-        } else {
-            $href = new moodle_url('/blocks/evasys_sync/sync.php');
-        }
 
-        // Initialize data for mustache template.
-        $startdisabled = false;
-        $enddisabled = false;
-        $emailsentnotice = false;
-        $periodsetnotice = false;
+        $evalrequest = \block_evasys_sync\evaluation_request::for_course($this->page->course->id);
+        $evaluations = \block_evasys_sync\evaluation::for_course($this->page->course->id);
 
-        // Set start to today and end to a week from now.
-        $start = time();
-        $oneweeklater = $time = new \DateTime();
-        $oneweeklater->add(new \DateInterval("P7D"));
-        $end = $oneweeklater->getTimestamp();
+        $stringformat = get_string('strftimedatetime', 'langconfig');
 
-        // See if there are any students that can evaluate.
-        // If there are no students we disable all controls.
-        $nostudents = (count_enrolled_users(
-                context_course::instance($this->page->course->id), 'block/evasys_sync:mayevaluate') == 0);
-
-        $recordhasstandardtime = false;
-
-        if ($record !== false) {
-            $state = $record->get('state');
-            if ($state == course_evaluation_allocation::STATE_MANUAL) {
-                // If the persistenceclass exists and the state is manual an email must have been sent.
-                $emailsentnotice = true;
-            }
-            if ($state == course_evaluation_allocation::STATE_AUTO_NOTOPENED) {
-                // If the persistenceclass exists and the state is automatic and not opened
-                // the period must have been set.
-                $periodsetnotice = true;
-            }
-            if ($state >= course_evaluation_allocation::STATE_AUTO_OPENED || $nostudents) {
-                // If the course was already opened, disable the start date. If there are no students disable all controls.
-                $startdisabled = true;
+        if ($evaluations) {
+            $this->content->text .= html_writer::tag('h5', get_string('evaluations', 'block_evasys_sync'));
```

Comment:
If an evaluation has been successfully requested, I would also add a corresponding message e.g. "Die Evaluation wurde beantragt für folgenden Zeitraum:"
A teacher chose the option that the course should not be evaluated, a corresponding message should also be shown for this scenario.